src/scn/BlockchainManager/BlockchainManager.h

```diff
@@ -42,22 +42,22 @@ namespace scn {
 
     class BlockchainManager {
     public:
-        BlockchainManager(public_key_t our_public_key,
-                private_key_t our_private_key,
+        BlockchainManager(const public_key_t& our_public_key,
+                const private_key_t& our_private_key,
                 Blockchain& blockchain,
                 IP2PConnector& p2p_connector,
                 IMiner& miner,
                 bool initial_fetch = true,
                 ISynchronizedTimer& sync_timer = static_sync_timer);
         virtual ~BlockchainManager();
 
-        virtual void baselineBlockReceivedCallback(const peer_id_t& peer_id, std::shared_ptr<const BaselineBlock> block, bool reply);
+        virtual void baselineBlockReceivedCallback(const peer_id_t& peer_id, const std::shared_ptr<const BaselineBlock>& block, bool reply);
 
```

Comment:
Thank you for the changes and the explanation regarding shared pointers. I am almost ready to merge it now :-)
Just one more thing here: The baselineBlockReceivedCallback in BlockchainManager still changes the shared_ptr to const share_ptr&, probably you just missed it by accident.
Thanks!